### DISCUSSION TITLE
RCP Openthread Border Router

### DISCUSSION THREAD

**Discussion by guerbasc** [openthread/openthread#7600] [Q&A]
Hi everyone,

I have configured RCP Silabs with  Rpi as Border router, i follow the steps detailing in openthread web site. " I got the ot-br-posix repository from github , i activated otbr-agent services and I have the ot-ctl application which manages OT1.2 " 
I have flashed silbas as RCP then plug it on the Rpi===> Silabs was successfully identified by thread test harness 
I do same configuration for NRF52840. I have flashed it as RCP then plug it on th Rpi ===>  RCP didn't identify by thread test harness

Are there any other steps to do for the  NRF52840  Border router other than what was described on the openthread site ?.
Thank you an advance 

Celia,

**Comment by jwhui** [comment]
@guerbasc , can you provide more details on:

1. Which nRF52840 hardware device are you using? USB dongle? DK? Something else?
2.  How you built the nRF52840 RCP image.
3.  How you connected the RCP to the RPi.
4. How you configured OTBR to connect to the nRF52840 RCP.

**Reply by guerbasc** [reply]
@jwhui  Hi,

I am using an nRF52840 USB dongle.
To build the nRF52840 RCP image, I used nRFutil.
To connect the RCP to the RPi, simply plug the dongel into the RPi.
To configure OTBR to connect the nRF52840, I followed the following steps:
• Attach and configure RCP device: To configure the RCP device's serial port in otbr-agent:
-Specify the serial port name for the RCP device by checking /dev: ls /dev/tty*
-Move into otbr-agent using cd /etc/default/
-Use the following configuration line: OTBR_AGENT_OPTS="-l wpan0 spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=100000"

(Ps: provided your RCP is using ttyACM0 if not change it to the predetermined serial port name).
• Verify that all required services are enabled: sudo systemctl status
• Reboot the Raspberry Pi.
• Verify that the RCP is in the correct state: sudo ot-ctl state
If the RCP is successfully running and the node is not a member of a Thread network, the output should be like the following: disabled

**Reply by jwhui** [reply]
Setting the UART baudrate to "100000" does not seem correct. If you are following the [OpenThread Border Router Build and Configuration Guide](https://openthread.io/guides/border-router/build), you should not need to update the `OTBR_AGENT_OPTS` configuration.

I'm still not clear on how you are building the RCP binary. Can you share exactly how you are building the RCP binary?

**Reply by guerbasc** [reply]
I find on the nRF connect website that it is necessary to set the baud rate to 1000000, then I change it to 1000000.

To build the RCP binary; I was using Nrfutil and windows command as described below : 

Make sure the device is in DFU bootloader mode – indicated by the pulsing red LED.

The device LED will start pulsing red, and the ‘USB Serial Device’ in Device Manager will be replaced by another ‘USB Serial Device’ with a different COM port. Be sure to note the new COM port number that appeared after the button push.

In your windows command, move into folder which contains firmware & nrfutil then use the following nrfutil  command (replace the folder with .zip extension with your firmware and set the port of your device which is displayed in the device manager)  :  
Example : 
C:\Users\guerbasc\Downloads\Nordic OT1.2> nrfutil dfu usb-serial -pkg ot-rcp-1.2-20220222-5c9a6f570.zip -p COM26 -b 115200

**Reply by jwhui** [reply]
Can you point to the nRF documentation that specifies baud rate of "1000000"?

Note that in https://github.com/openthread/openthread/discussions/7600#discussioncomment-2600013, you specified a baud rate of "100000", which is not the same as "1000000".

**Reply by guerbasc** [reply]
I was wrong I wanted to write 1000000 in  #7600 

You can see in the following nRF web Site that they specified this baudrate . 
https://developer-nordicsemi-com.translate.goog/nRF_Connect_SDK/doc/latest/nrf/ug_thread_tools.html?_x_tr_sl=en&_x_tr_tl=fr&_x_tr_hl=fr&_x_tr_pto=op,sc#running-the-openthread-posix-applications

if you have another method to configure Nrf dongle border router, I'm interested !

**Reply by jwhui** [reply]
Maybe @edmont can provide more insight?

**Reply by guerbasc** [reply]
hello @edmont, do you have any specific recommendations for setting up nRF52840 as a border router that you could provide me please!

**Reply by edmont** [reply]
Hi @guerbasc, I notice that you are using `ot-rcp-1.2-20220222-5c9a6f570.zip`, a dongle firmware provided by Nordic for Thread Certification reference device. Are you using the matching Raspberry Pi image? Should be `otbr.20220222.img.zip`.

By the way, this kind of questions about Thread Test Harness should be made preferably in the Thread Group's Jira platform.

**Reply by guerbasc** [reply]
@edmont, thank you for  you prompt response :)

Yes i use ot-rcp-1.2-20220222-5c9a6f570.zip firmware.

Where could i found otbr.20220222.img.zip, a OTBR  Raspberry Pi image ?

**Reply by edmont** [reply]
Access to the Thread Harness is restricted to some Thread Group companies depending on their membership level: https://www.threadgroup.org/thread-group#Membershipbenefits

**Reply by guerbasc** [reply]
Thank's for help @edmont @jwhui
